src/main/java/com/salesforce/einsteinbot/sdk/client/SessionManagedChatbotClient.java

```diff
@@ -133,8 +135,9 @@ private void addInitMessage(RequestEnvelope requestEnvelope) {
         addIntegrationTypeAndNameToContextVariables(initMessage.getVariables(), integrationName));
 
     //replace first message with init message
-    requestEnvelope.getMessages().set(0, initMessage);
-
+    List<AnyRequestMessage> messages = new ArrayList<>(requestEnvelope.getMessages());
```

Comment:
May be worth adding a test where we pass `Collections.singletonList` as messages in request envelope and that doesn't cause any errors.

Reply:
Updated existing test to use `Collections.singletonList` , trying to not spend too much effort on this as this code doesn't exist in v5.